src/Gaufrette/Adapter/AzureBlobStorage.php

```diff
@@ -120,78 +156,91 @@ public function deleteContainer($containerName, DeleteContainerOptions $options
 
     /**
      * {@inheritdoc}
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
      */
     public function read($key)
     {
         $this->init();
+        list($containerName, $key) = $this->tokenizeKey($key);
 
         try {
-            $blob = $this->blobProxy->getBlob($this->containerName, $key);
+            $blob = $this->blobProxy->getBlob($containerName, $key);
 
             return stream_get_contents($blob->getContentStream());
         } catch (ServiceException $e) {
-            $this->failIfContainerNotFound($e, sprintf('read key "%s"', $key));
+            $this->failIfContainerNotFound($e, sprintf('read key "%s"', $key), $containerName);
 
             return false;
         }
     }
 
     /**
      * {@inheritdoc}
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
      */
     public function write($key, $content)
     {
         $this->init();
+        list($containerName, $key) = $this->tokenizeKey($key);
 
-        try {
-            $options = new CreateBlobOptions();
-
-            if ($this->detectContentType) {
-                $contentType = $this->guessContentType($content);
+        $options = new CreateBlobOptions();
 
-                $options->setBlobContentType($contentType);
-            }
+        if ($this->detectContentType) {
+            $contentType = $this->guessContentType($content);
 
-            $this->blobProxy->createBlockBlob($this->containerName, $key, $content, $options);
+            $options->setContentType($contentType);
+        }
 
-            if (is_resource($content)) {
-                return Util\Size::fromResource($content);
-            }
+        try {
+            $this->createContainer($containerName);
```

Comment:
Multi-container mode ignores `$create` as it wouldn't make much sense to say "Create containers automatically, but don't". :) Agree to check for multi-mode here, will do it today. 👍 

Thanks, it's for my selfish interest to have this feature, obviously, but glad there's a Gaufrette to contribute it to in the first place, thank you all for that.